### ISSUE TITLE
Alienware 15 R4 with Intel Arc A770..

### ISSUE THREAD

**Issue by chaoshusky** [Shidell/OpenCaldera#15]
No idea if this project is still alive and i have since removed the card. I was able to run my Arc A770 LE with the AGA, however as it needs ReBAR to perform well it scored quite low on Time Spy and would likely be rather limited by the x4 link speed if the BIOS had it enabled (seems it reports this laptop could do it, but no BIOS option exposed to enable it..most likely because it has a GTX 1070 internally lol) unless someone would finally unlock the BIOS in this laptop and also remove the plundervolt 'fix' from the latest one so i can still undervolt the CPU..

Anyway just thought i'd chime in with that as i can't imagine anyone tried one in it! The 1660 Super the amp came with works fine of course but isn't much better than the onboard.. And A Radeon Vega 56 didn't fit but tested it with the case open and it absolutely HATED that card (Powercolor Red Dragon) though did eventually get it working by plugging the AGA in when the laptop was booted - it refused to work from a cold boot! Most likely a CPU and x4 link limit, sending the picture back to the built in screen, but most cards seem to score about the same as the onboard or slightly beat it - i have a 2080 Super in there now and that scores close to the A770.. Though that Intel card in a full PC behind me scores 14K on Time Spy instead of just over 9K!

**Comment by Shidell** [comment]
Yeah, unfortunately without ReBAR, Arc isn't viable. There is a GitHub project to expose (enable) ReBAR on motherboards that don't support it, and although I played with my Alienware Area 51m R2 bios (and theoretically "enabled" ReBAR), I'm not willing to flash it (at risk of bricking it.) If you're really adventurous, you can look into it yourself, but I'd caution you that it's at your own risk and arguably, pretty dangerous.

Most people have delidded their AGA (myself included) to accommodate newer GPUs. Sorry to hear your experience with your Vega 56 wasn't positive; in general, you should be able to use one and get full performance (up to and/or including even an RX 7000 series.) You'll probably run into CPU driver overhead with a GeForce, which might cannibalize performance some.

With a 7900 XTX in my A51MR2, I managed ~23k in Time Spy (~28k graphics, ~12k cpu score.) The 4x PCIE limit doesn't really hinder much, unless your GPU has low VRAM (8 GB or less.) Also, the higher the resolution, the less it's a factor.

**Comment by chaoshusky** [comment]
Ah, sorry to hear you ended up with an AMD card, i only needed it for a Hackintosh.. If i couldn't use or didn't have an nVidia card i'd much rather use my Arc! Some AMD gamer girl fanatic sold me the Vega 56 and surprise surprise its defective. Tends to overheat and crash if not extremely careful - seems she ran it in 'Fan Stop' mode all its life and it sat there cooking at 50C idle.. My 10 year old GTX 970 on the other hand is still fine! Haha.. Driver overhead or not i don't want to deal with Radeon drivers!

I'd be up for trying a ReBAR or even full BIOS unlock mod if it can be based on a newer one, using an older BIOS seems to be causing a weird issue that causes the laptop to briefly freeze what's running, sound slows down, even if not using the AGA then resumes working. Seems random too, might happen a couple of times in half an hour or once in a few hours lol though i'm running an older BIOS to keep the ability to undervolt the CPU as it'll still peak at 96C with a -100mv undervolt, toasty! Intel gear can handle it though, at least there's that. I can't believe that's all a 7900XTX gets on Time Spy! The Arc card manages over 14K and my old faithful 3090 still hits a tad over 21K.. Strange. What CPU do you use to get a score that low? Don't mean to sound like a d*ck, just curious! I get just over 18k CPU score on a 9980XE and 12900K.. I do have an external programmer or two so hopefully i wouldn't brick my laptop, but the recovery mode would save it so long as any edits don't damage that area.

As for the 4x link not affecting anything, it will definitely affect ReBAR, you'll only be able to transfer between system and video RAM at 25% of the speed lol

And some seriously "de-lid" the AGA? I did that to my 8086K hah but that..would be ugly as heck. Fortunately a 2080 Super founders fits in there perfectly, runs fairly cool and I'm only using 1080p on there anyway. I'd much rather use a card that fits or change the card's cooling than do that!

**Comment by Shidell** [comment]
I chose the Radeon specifically, because the driver overhead with a GeForce cannibalizes performance. There's no point in purchasing a 3090, for example, if it's limited to performing like a 3070 due to the CPU be unable to drive it further.

You need to keep in mind that most gaming laptops are using mobile CPUs, running at 45w (as compared to 125w for the desktop equivalent.) For almost every Alienware system available, the CPUs are already a little "older", because Alienware stopped including the AGA port on newer hardware.

A 7900 XTX in the highest-end desktop system available will obviously score much higher; again, we need to keep in mind that we're comparing against older systems—slower CPUs, slower memory, slower busses.

And yeah, many remove the lid of the AGA. You can unlock more performance with a newer card, but you'll need to remove the lid for it to fit.
